### ISSUE TITLE
Add `deno resolve <path>` to power community bundlers

### ISSUE THREAD

**Issue by StarLederer** [denoland/deno#20493]
# Motivation

It looks like Deno currently has a limited ecosystem when it comes to bundling for production.

`deno bundle` is deprecated and does not support NPM. `deno_emit` is not finished yet. `rollup` and `esbuild` are mentioned in the docs but I could not find up-to-date Dono plugins or bootstrap configuration for either.

I would like to give building a Rollup plugin a try but I cannot imagine being able to maintain it so I hope It will turn out to be small enough to just live in the Rollup config.

From what I gathered so far, my rollup config simply needs to:
* resolve imports aliased using import maps,
* use SWC for everything else.

In this issue I would like to request a tool that makes the former very easy to implement. For small DIY Rollup configs, and big third-party bundlers that might come in the future.

# Description

I would like to request a `resolve` subcommand added to Deno that takes a module import path as an argument and outputs the absolute location of that module on the file system (potentially in the Deno cache).

For example:

```js
/**
 * @module /app/mod.js does not affect anything
 * in this example but demonstrates where the
 * module imports that we need to resolve would
 * normally be found.
 */
import foo from "./bar.js";
import fizz from "@buzz";
import http from "https://example.com/http.js";
import letfpad from "npm:left-pad";
```

```sh
$ pwd
> /app

$ deno resolve "/app/mod.js"
> /app/mod.js

$ deno resolve "./bar.js"
> /app/bar.js

$ deno resolve "@buzz"
> /app/buzz/mod.js

$ echo $DENO_DIR
> /deno

$ deno resolve "https://example.com/http.js"
> /deno/gen/https/example.com/123abc.js

$ deno resolve "npm:left-pad"
> /deno/npm/registry.npmjs.org/left-pad/6.9.0/dist/index.js
```

Of course, this command should respect the `deno.json` and `deno.lock` files and resolve dependencies based on the working direcotry.

**Comment by lucacasonato** [comment]
`deno info <specifier> --json`

**Comment by QuentinRoy** [comment]
@StarLederer, where you able to build said rollup plugin? I'm running on the same issue, and couldn't find a way to make rollup to resolve deno modules...

**Comment by StarLederer** [comment]
@QuentinRoy I was but with no NPM support. I concluded it was too much for a quick side-project and never fished the plugin.

This is very uncomfortable to say here but I have completely given up on Deno. I am extremely grateful for the innovation that is has driven but now that Node has adopted the good ideas from it, and the project demonstrated that just being written in Rust doesn't mean development goes faster (e.g. [we still don't have a working LSP](https://github.com/denoland/deno/issues/13404)), I have returned to Node.

Luckily, I have, at some point, stumble upon a project that had the same aim that seemed to be going very strong. I don't remember what it was but I suspect it was [esbuild_deno_loader](https://github.com/lucacasonato/esbuild_deno_loader) by our friendly Deno maintainer that answered this thread. I think that should be able to solve your issue.
